### PR TITLE
Handle configuration validation and plugin installation in a Vagrant plugin

### PR DESCRIPTION
**Pull request by vdloo**
bundle vagrant-hypconfigmgmt plugin
--------------------------------------------------------------------------------

The basic hypernode-vagrant box can now be configured and booted by cloning the repo and running "vagrant up" three times and following the dialogue. Vagrant needs to be started three consecutive times because it loads all configurations during initialization, not during runtime. When we change the configuration in the Vagrant process we need to restart Vagrant for the changes to take effect.

1. First vagrant up: Install the vagrant-hypconfigmgmt plugin
2. Second vagrant up: Configure the local.yml and install other required plugins
3. Third vagrant up: Boot the box

When the box is properly configured and all the required plugins is installed one vagrant up is sufficient so this will have no impact if you (correctly) manually manage your local.yml file. Prompting for options can be disabled by removing `config.hypconfigmgmt.enabled = true` in the Vagrantfile (but the config validation will also be gone so beware).

Fresh install:
bootvagrant
What the configuration file will look like after completing the dialogue:
localyml

Running "vagrant up" again after only changing the Magento version from 2 to 1 in the local.yml configuration file.
bootvagrant2
The resulting local.yml (note the disabled mounted directory)
localyml2

tldr: simplified installation to calling "vagrant up" three times. settings are prompted during "vagrant up". magento1 and magento2 mount paths are swapped out when the version is changed in the config. Platform for moving complexity out of the Vagrantfile and into the bundled plugin.

### CODE REVIEW

**Comment by vdloo** [comment]
blocked because of an issue with vagrant 1.8.4 see https://github.com/mitchellh/vagrant/issues/7493

**Comment by vdloo** [comment]
@hongaar @timneutkens what do you think?

**Comment by hongaar** [comment]
Sweet. :+1: for a cleaner Vagrantfile and offloading configuration to a separate plugin.

**Comment by timneutkens** [comment]
Cool stuff, seems like a good solution! 👍
